### PR TITLE
fixes #51

### LINKED ISSUE #51: execution launch fails if port is an empty string
should handle the case where context is:

```
{
  "image": "hello-world"
  "ports": [""]
}

### PR DESCRIPTION
**Pull request by rokroskar**
also skips k8s ingress in tests

### CODE REVIEW

File: tests/test_deployer.py [hunk 1]
```diff
@@ -46,8 +46,13 @@ def test_execution_launch(app, engine, deployer):
     assert isinstance(context, Context)
     assert isinstance(execution, Execution)
     assert execution.engine == engine
+    assert 'Hello from Docker!' in deployer.get_logs(execution)
 
+    # test that image deployment works with a variety of ports options
+    context = deployer.create({'image': 'hello-world', 'ports': ['', '9999']})
```
Comment (jirikuncar) [inline tests/test_deployer.py]: we can `parametrize` the spec json

Comment (jirikuncar) [reply tests/test_deployer.py]:
```python
@pytest.mark.parametrize('engine', ['docker', 'k8s'])
@pytest.mark.parametrize('spec', [
    {'image': 'hello-world'},
    {'image': 'hello-world', 'ports': ['', '9999']},
])
def test_execution_launch(app, engine, spec, deployer):
```

File: renga_deployer/engines.py [hunk 2]
```diff
@@ -206,7 +207,8 @@ def launch(self, execution, engine=None, **kwargs):
                 extra={'service': service.to_dict()})
 
             # if using an ingress, need to make an additional object
-            if current_app.config.get('DEPLOYER_K8S_USE_INGRESS'):
+            if current_app.config.get(
+                    'DEPLOYER_K8S_USE_INGRESS'):  # pragma no cover
```
Comment (jirikuncar) [inline renga_deployer/engines.py]: 😄

File: renga_deployer/engines.py [hunk 3]
```diff
@@ -240,7 +242,8 @@ def stop(self, execution, remove=False):
                 'Deleted namespaced service {}'.format(service.metadata.name),
                 extra={'service': service.to_dict()})
 
-            if current_app.config.get('DEPLOYER_K8S_USE_INGRESS'):
+            if current_app.config.get(
+                    'DEPLOYER_K8S_USE_INGRESS'):  # pragma no cover
```
Comment (jirikuncar) [inline renga_deployer/engines.py]: 😄

File: renga_deployer/engines.py [hunk 4]
```diff
@@ -347,7 +350,8 @@ def _k8s_service_template(namespace, context, uid):
         }
 
     @staticmethod
-    def _k8s_ingress_template(uid, service_name, service_port):
+    def _k8s_ingress_template(
+            uid, service_name, service_port):  # pragma no cover
```
Comment (jirikuncar) [inline renga_deployer/engines.py]: 😄

File: renga_deployer/engines.py (hunk 3 above)
Comment (rokroskar) [reply renga_deployer/engines.py]: I think you mean🥇

**Pull request merged**